### ISSUE TITLE
[Clouds] Support paperspace

### ISSUE THREAD

**Issue by fakerybakery** [skypilot-org/skypilot#3077]
Hi,
Is Paperspace supported?
Thank you!

**Comment by Michaelvll** [comment]
Thanks for asking @fakerybakery! We don’t have the support for paperspace at the moment, but we do have a community member trying to add it, cc’ing @asaiacai for a look. ;)

**Comment by fakerybakery** [comment]
Thanks! I was hoping to be able to use it in Gradient, for example running some code in the notebook to allow Skypilot to connect to it?

**Comment by asaiacai** [comment]
Hi @fakerybakery! For context, I think I've figured out all the APIs for machine create/up/stop/down to integrate into skypilot. Recently I was stuck on how to add `sky-key.pub` to the machines on creation but recently I found out that [paperspace provides an API for startup scripts upon machine creation](https://docs.paperspace.com/core/api-reference/scripts) that I'm going to try and use for adding the public keys. I'm planning on drafting a PR sometime this month. Stay tuned!
